google/cloud/spanner/internal/connection_impl_test.cc

```diff
@@ -324,7 +334,8 @@ TEST(ConnectionImplTest, ReadCreateSessionFailure) {
       .WillRepeatedly(Return(Status(StatusCode::kPermissionDenied,
                                     "uh-oh in BatchCreateSessions")));
 
-  auto conn = MakeConnectionImpl(db, {mock});
+  internal::OptionsSpan span(MakeLimitedTimeOptions());
+  auto conn = MakeConnectionImpl(db, mock);
```

Comment:
I agree that what you suggest is nicer. I think only 2 is necessary though. I discovered that the connection options in this test are only used for `SessionPool` business, and `MakeLimitedRetryOptions()` works for all the cases.

Reply:
>> It seems like it might be nicer to: 1. restore the `Options` argument to `MakeConnectionImpl()`
> 
> I agree that what you suggest is nicer. I think only 2 is necessary though. I discovered that the connection options in this test are only used for `SessionPool` business, and `MakeLimitedRetryOptions()` works for all the cases.

Yes, 1 is not necessary until we need to set a `SessionPool` option in a test, which a pending change I have does need.  But I forgot to emphasize that, so I'll just restore the ability in that pending change.